scarlet/initialization.py

```diff
@@ -0,0 +1,438 @@
+from .source import *
+from .wavelet import Starlet
+from .observation import Observation, LowResObservation
+from .interpolation import interpolate_observation
+from . import operator
+from .bbox import Box
+
+import numpy as np
+
+import logging
+
+logger = logging.getLogger("scarlet.initialisation")
+
+def trim_morphology(sky_coord, frame, morph, bg_cutoff, thresh):
+    # trim morph to pixels above threshold
+    mask = morph > bg_cutoff * thresh
+    boxsize = 16
+    pixel_center = frame.get_pixel(sky_coord)
+    if mask.sum() > 0:
+        morph[~mask] = 0
+
+        # normalize to unity at peak pixel
+        center_morph = morph[pixel_center[0], pixel_center[1]]
+        morph /= center_morph
+
+        # find fitting bbox
+        bbox = Box.from_data(morph, min_value=0)
+        if bbox.contains(pixel_center):
+            size = 2 * max(
+                (
+                    pixel_center[0] - bbox.start[-2],
+                    bbox.stop[0] - pixel_center[-2],
+                    pixel_center[1] - bbox.start[-1],
+                    bbox.stop[1] - pixel_center[-1],
+                )
+            )
+            while boxsize < size:
+                boxsize *= 2
+    else:
+        msg = "No flux above threshold for source at y={0} x={1}".format(*pixel_center)
+        logger.warning(msg)
+
+    # define bbox and trim to bbox
+    bottom = pixel_center[0] - boxsize // 2
+    top = pixel_center[0] + boxsize // 2
+    left = pixel_center[1] - boxsize // 2
+    right = pixel_center[1] + boxsize // 2
+    bbox = Box.from_bounds((bottom, top), (left, right))
+    morph = bbox.extract_from(morph)
+    bbox_3d = Box.from_bounds((0, frame.C), (bottom, top), (left, right))
+    return morph, bbox_3d
+
+
+def init_extended_source(
+    sky_coord,
+    frame,
+    observations,
+    coadd=None,
+    bg_cutoff=None,
+    thresh=1,
+    symmetric=True,
+    monotonic=True,
+    min_grad=0.2,
```

Comment:
Please make the default `min_grad=0.0`, since we know that higher values do not work well for many objects with the new `flat` monotonicity weighting.

Also, please change `monotonic` to allow the weight type to be specified, the same as `ExtendedSource`. This is make sure that our initialization and constraints are consistent.

Reply:
Done